src/main.rs

```diff
@@ -94,10 +94,40 @@ fn default_filename(filename: Option<String>) -> Result<PathBuf, clap::error::Er
     ))
 }
 
+fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Error> {
+    // Extract the filename itself, as well as the directory from `path`.
+    assert!(path.is_file());
+    let filename_without_path = path.file_name().unwrap().to_str().unwrap();
+    let directory = path.parent().unwrap().to_str().unwrap();
+
+    let output = std::process::Command::new("git")
+        .arg("-C")
+        .arg(directory)
+        .arg("status")
+        .arg("--porcelain")
+        .arg(filename_without_path)
+        .output()
+        .expect("Failed to run `git status");
+
+    let git_status = String::from_utf8_lossy(&output.stdout);
+
+    // This implies that the spec we're targeting has no uncommitted changes,
+    // and so we're safe to proceed with rewrapping.
+    if !git_status.contains(&filename_without_path) {
```

Comment:
You can just check if `output.stdout` is non-empty now, since the filename filter means it can only contain the one file.
```suggestion
    // This implies that the spec we're targeting has no uncommitted changes,
    // and so we're safe to proceed with rewrapping.
    if output.stdout.is_empty() {
```